lib/amqp/channel.ex

```diff
@@ -17,7 +17,8 @@ defmodule AMQP.Channel do
   `:amqp_connection.open_channel/2` for more details and `AMQP.DirectConsumer` for an example of a
   custom consumer.
   """
-  @spec open(Connection.t(), custom_consumer | nil) :: {:ok, Channel.t()} | {:error, any}
+  @spec open(Connection.t(), custom_consumer | nil) ::
+          {:ok, Channel.t()} | {:error, any} | any
```

Comment:
Again, wanted to keep the tuple variant, just in case, but if the signature here needs to change, let me know.